src/site/content/en/blog/shape-detection/index.md

```diff
@@ -0,0 +1,351 @@
+([bug](https://crbug.com/920961)). As a temporary measure, we instead
+recommend a *defensive programming* approach by doing feature detection
+like this:
+
+```js
+const supported = await (async () => 'FaceDetector' in window &&
+    await new FaceDetector().detect(document.createElement('canvas'))
+    .then(_ => true)
+    .catch(e => e.name === 'NotSupportedError' ? false : true))();
+```
+
+## Best practices {: #bestpractices}
+
+All detectors work asynchronously, that is, they do not block the main
+thread. So don't rely on realtime detection, but rather allow for some
+time for the detector to do its work.
+
+If you are a fan of
+[Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API),
+you'll be happy tro know thatß detectors are exposed there as well.
+Detection results are serializable and can thus be passedß from the worker
+to the main app via `postMessage()`. The [demo][demo] shows this in action.
+
+Not all platform implementations support all features, so be sure to check
+the support situation carefully and use the API as a progressive
+enhancement. For example, some platforms might support face detection
+per se, but not face landmark detection (eyes, nose, mouth, etc.); or the
+existence and the location of text may be recognized, but not text contents.
+
+{% Aside 'caution' %}
+  This API is an optimization and not something guaranteed to be available
+  from the platform for every user. Developers are expected to combine this
+  with their own [image recognition code](https://github.com/mjyc/opencv) and
+  take advantage of the native optimization when it is available.
+{% endAside %}
+
+## Feedback {: #feedback }
+
+We want to hear about your experiences with the Shape Detection API.
+
+### Tell us about the API design {: .hide-from-toc }
+
+Is there something about the API that doesn't work like you expected? Or
+are there missing methods or properties that you need to implement your
+idea? Have a question or comment on the security model?
+
+* File a spec issue on the [Shape Detection API GitHub repo][issues],
+  or add your thoughts to an existing issue.
+
+### Problem with the implementation? {: .hide-from-toc }
+
+Did you find a bug with Chrome's implementation? Or is the implementation
+different from the spec?
+
+* File a bug at <https://new.crbug.com>. Be sure to include as
+  much detail as you can, simple instructions for reproducing, and set
+  *Components* to `Blink>ImageCapture`. [Glitch](https://glitch.com)
+  works great for sharing quick and easy repros.
+
+### Planning to use the API? {: .hide-from-toc }
+
+Planning to use the Shape Detection API on your site? Your public support
+helps us to prioritize features, and shows other browser vendors how
+critical it is to support them.
+
+* Share how you plan to use it on the [WICG Discourse thread][wicg-discourse]
+* Send a Tweet to [@ChromiumDev][cr-dev-twitter] with `#shapedetection` and
+  let us know where and how you're using it.
```

Comment:
@robdodson authors would have to pass in a lot of data but all of this content can theoretically go in a component. The only things that change are the API name, the spec repo, the Blink component, etc.